lib/client.js

```diff
@@ -50,28 +49,98 @@ class CloudantClient {
     // build plugin array
     var plugins = [];
     if (typeof this._cfg.plugin === 'undefined' && typeof this._cfg.plugins === 'undefined') {
-      plugins = ['cookieauth']; // default
+      plugins = [ 'cookieauth' ]; // default
     } else {
       [].concat(self._cfg.plugin).concat(self._cfg.plugins).forEach(function(plugin) {
-        if (typeof plugin === 'function' && !(plugin._pluginVersion >= 2)) {
-          if (self.useLegacyPlugin) {
-            throw new Error('Using multiple legacy plugins in not supported');
-          } else {
-            debug('Using legacy plugin.');
-            self.useLegacyPlugin = true;
-            client = plugin; // use legacy plugin as client
-            return;
-          }
+        if (typeof plugin !== 'function' || plugin.pluginVersion >= 2) {
+          plugins.push(plugin);
+        } else if (self.useLegacyPlugin) {
+          throw new Error('Using multiple legacy plugins in not supported');
```

Comment:
`in` -> `is`

and perhaps we should say `permitted` rather than `supported` given the latter may feel a bit like "you can try if you want, but you're on your own", but the former is the more correct "we won't allow you to use this configuration".

Reply:
Done.